encoding/kzg/prover/prover.go

```diff
@@ -66,12 +66,39 @@ func NewProver(kzgConfig *kzg.KzgConfig, encoderConfig *encoding.Config) (*Prove
 			return nil, err
 		}
 
-		g2Trailing, err = kzg.ReadG2PointSection(
-			kzgConfig.G2Path,
-			kzgConfig.SRSOrder-kzgConfig.SRSNumberToLoad,
-			kzgConfig.SRSOrder, // last exclusive
-			kzgConfig.NumWorker,
-		)
+		hasG2TrailingFile := len(kzgConfig.G2TrailingPath) != 0
+		if hasG2TrailingFile {
+			fileStat, err := os.Stat(kzgConfig.G2TrailingPath)
+			if err != nil {
+				return nil, err
+			}
+			fileSizeByte := fileStat.Size()
+			if fileSizeByte%64 != 0 {
+				return nil, errors.New("corrupted g2 points from G2TrailingPath.")
+			}
+			// get the size
+			numG2point := uint64(fileSizeByte / kzg.G2PointBytes)
+			if numG2point < kzgConfig.SRSNumberToLoad {
+				return nil, errors.New("Insufficent number of g2 points from G2TrailingPath")
```

Comment:
more info would be helpful here, too. how many g2 points exist, vs how many are being attempted to load?